### PR TITLE
fix: replace colon with pipe separator for Windows path [IDE-1300]

### PR DESCRIPTION
**Pull request by bastiandoetsch**
### Description
- Changed navigation separator from ':' to '|' to fix navigation issues on Windows
- Colons in Windows drive paths (e.g., C:) were causing incorrect string splitting
- Added comprehensive unit and integration tests for the separator change
- Tests cover Windows paths, URLs with colons, and edge cases

_Provide description of this PR and changes, if linked Jira ticket doesn't cover it in full._

### Checklist

- [x] Tests added and all succeed
- [x] Linted
- [x] CHANGELOG.md updated
- [ ] README.md updated, if user-facing

### Screenshots / GIFs

_Visuals that may help the reviewer. Please add screenshots for any UI change. GIFs are most welcome!_

### CODE REVIEW

File: src/main/kotlin/io/snyk/plugin/ui/jcef/OpenFileLoadHandlerGenerator.kt [hunk 1]
```diff
@@ -25,9 +27,18 @@ class OpenFileLoadHandlerGenerator(
         val virtualFile = virtualFiles[filePath] ?: return JBCefJSQuery.Response("success")
 
         val document = virtualFile.getDocument()
-        val startLineStartOffset = document?.getLineStartOffset(startLine) ?: 0
-        val startOffset = startLineStartOffset + (startCharacter)
-        val endLineStartOffset = document?.getLineStartOffset(endLine) ?: 0
+        if (document == null) {
```
Comment (andrewrobinsonhodges-snyk) [inline src/main/kotlin/io/snyk/plugin/ui/jcef/OpenFileLoadHandlerGenerator.kt:30]: nitpick - style/idiomaticity: could combine with line 29 and use an Elvis operator as on line 27, for consistency.

File: src/main/kotlin/io/snyk/plugin/ui/jcef/OpenFileLoadHandlerGenerator.kt [hunk 2]
```diff
@@ -10,12 +10,14 @@ import org.cef.browser.CefBrowser
 import org.cef.browser.CefFrame
 import org.cef.handler.CefLoadHandlerAdapter
 
+const val navigationSeparator = "|"
```
Comment (andrewrobinsonhodges-snyk) [inline src/main/kotlin/io/snyk/plugin/ui/jcef/OpenFileLoadHandlerGenerator.kt:13]: Might be worth a comment explaining why we chose a vertical pipe.

**Review by andrewrobinsonhodges-snyk — APPROVED** [review]
